crates/burn-import/src/burn/node/range.rs

```diff
@@ -0,0 +1,118 @@
+use super::{Node, NodeCodegen};
+use crate::burn::{Scope, TensorType, Type};
+use burn::record::PrecisionSettings;
+use proc_macro2::TokenStream;
+use quote::quote;
+
+#[derive(Debug, Clone, new)]
+pub struct RangeNode {
+    pub start: Type,
+    pub end: Type,
+    pub step: Type,
+    pub output: TensorType,
+}
+
+impl<PS: PrecisionSettings> NodeCodegen<PS> for RangeNode {
+    fn output_types(&self) -> Vec<Type> {
+        vec![Type::Tensor(self.output.clone())]
+    }
+
+    fn input_types(&self) -> Vec<Type> {
+        vec![self.start.clone(), self.end.clone(), self.step.clone()]
+    }
+
+    fn forward(&self, _scope: &mut Scope, _node_position: usize) -> TokenStream {
+        let output = &self.output.name;
+
+        let start = match &self.start {
+            Type::Scalar(s) => {
+                let name = s.name.clone();
+                quote! { #name }
+            }
+            _ => panic!("Start must be a scalar"),
+        };
+
+        let end = match &self.end {
+            Type::Scalar(s) => {
+                let name = s.name.clone();
+                quote! { #name }
+            }
+            _ => panic!("End must be a scalar"),
+        };
+
+        let step = match &self.step {
+            Type::Scalar(s) => {
+                let name = s.name.clone();
+                quote! { #name }
+            }
+            _ => panic!("Step must be a scalar"),
+        };
```

Comment:
If the three fields are all supposed to be scalar, I suggest we change the node fields to:

```rust
pub struct RangeNode {
    pub start: ScalarType,
    pub end: ScalarType,
    pub step: ScalarType,
    pub output: TensorType,
}
```

and the logic to check the extract the scalar types can go in the `range_conversion`.

Reply:
Well yes, scalars are simply represented as 0-dim tensors. Basically what I am suggesting is just to move the type handling outside of the forward node method to the `range_conversion` function instead, a bit like this:

```rust
  fn range_conversion(node: Node) -> RangeNode {
      let output = node.outputs.first().unwrap().to_tensor_type();
      let start = node.inputs.first().unwrap().to_type();
      let end = node.inputs.get(1).unwrap().to_type();
      let step = node.inputs.get(2).unwrap().to_type();

      // Check/convert start, end & step to ScalarType just as it was done in the forward method
     // ...

      RangeNode::new(start, end, step, output)
  }
```